sopel/irc/backends.py

```diff
@@ -415,8 +479,24 @@ async def _run_forever(self) -> None:
         self._read_task = asyncio.create_task(self.read_forever())
         try:
             await self._read_task
+
+        # task was cancelled, i.e. another exception is responsible for that
         except asyncio.CancelledError:
             LOGGER.debug('Read task was cancelled.')
+
+        # connection reset requires a log, but no exception log
+        except ConnectionResetError as err:
+            LOGGER.error('Connection reset on read: %s', err)
+
+        # generic (connection) error requires a specific exception log
+        except ConnectionError as err:
+            LOGGER.error('Connection error on read: %s', err)
+            self.log_exception()
+        except Exception as err:
```

Comment:
```suggestion

        except Exception as err:
```

There's no good reason for this, except that it looks weird to me when every other `except` block gets its own pre-gap.

Reply:
Ha! That's because the `# generic (connection) error` comment above applies to both exception! Yet I understand why you would prefer with a blank line here.